tui/actions.go

```diff
@@ -112,30 +113,56 @@ func (ui *UI) AnalyzePath(path string, parentDir *analyze.Dir) error {
 	return nil
 }
 
-func (ui *UI) deleteSelected() {
+func (ui *UI) deleteSelected(isEmpty bool) {
 	row, column := ui.table.GetSelection()
-	selectedFile := ui.table.GetCell(row, column).GetReference().(analyze.Item)
-
-	modal := tview.NewModal().SetText("Deleting " + selectedFile.GetName() + "...")
-	ui.pages.AddPage("deleting", modal, true, true)
+	selectedItem := ui.table.GetCell(row, column).GetReference().(analyze.Item)
 
-	currentDir := ui.currentDir
+	var action, acting string
+	if isEmpty {
+		action = "empty "
+		acting = "emptying"
+	} else {
+		action = "delete "
+		acting = "deleting"
+	}
+	modal := tview.NewModal().SetText(strings.Title(acting) + " " + selectedItem.GetName() + "...")
+	ui.pages.AddPage(acting, modal, true, true)
+
+	var currentDir *analyze.Dir
+	var currentItems []analyze.Item
+	if isEmpty && selectedItem.IsDir() {
+		currentDir = selectedItem.(*analyze.Dir)
+		for _, file := range currentDir.Files {
+			currentItems = append(currentItems, file)
+		}
+	} else {
+		currentDir = ui.currentDir
+		currentItems = append(currentItems, selectedItem)
+	}
 
 	go func() {
-		if err := ui.remover(currentDir, selectedFile); err != nil {
-			msg := "Can't delete " + selectedFile.GetName()
-			ui.app.QueueUpdateDraw(func() {
-				ui.pages.RemovePage("deleting")
-				ui.showErr(msg, err)
-			})
-			if ui.done != nil {
-				ui.done <- struct{}{}
+		for _, item := range currentItems {
+			var err error
+			if isEmpty && !selectedItem.IsDir() {
```

Comment:
I think you could do this check just once outside the loop and store the function (emptier or remover) into a variable to be executed here.

Reply:
Thanks for your excellent code suggestions.